### PR TITLE
[Merged by Bors] - chore: bump std

### PR DESCRIPTION
**Pull request by kim-em**
Some deleted lemmas have been upstreamed to Std.

Note that the statements of `List.zipWith_map_left` (and `_right`) have been changed, requiring slight changes here.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Lists.lean [hunk 1]
```diff
@@ -437,6 +437,10 @@ termination_by
 #align lists.subset.decidable Lists.Subset.decidable
 #align lists.mem.decidable Lists.mem.decidable
 
+-- This is otherwise detected by the `nonClassInstance` linter from
+-- https://github.com/leanprover/std4/pull/268
+attribute [-instance] Lists.Equiv.decidable._unary._mutual
```
Comment (semorrison) [inline Mathlib/SetTheory/Lists.lean:442]: @alexjbest do you undestand what is happening here?

(Let's not wait for a resolution to this before merging this PR.)

Comment (alexjbest) [reply Mathlib/SetTheory/Lists.lean]: I think its simply that when you make a mutual `def/theorem/instance` lean creates the internal autogenerated mutual parts as `def/theorem/instance` correspondingly, which seems non-ideal in the `instance` cases but not high priority to change as mutual instances are quite rare.

**Comment by alexjbest** [comment]
maintainer merge

File: Mathlib/SetTheory/Lists.lean [hunk 2]
```diff
@@ -437,6 +437,10 @@ termination_by
 #align lists.subset.decidable Lists.Subset.decidable
 #align lists.mem.decidable Lists.mem.decidable
 
+-- This is otherwise detected by the `nonClassInstance` linter from
+-- https://github.com/leanprover/std4/pull/268
```
Comment (eric-wieser) [inline Mathlib/SetTheory/Lists.lean:441]:
```suggestion
-- This is an autogenerated declaration, so there's nothing we can do about it.
```
I don't think the comment needs to explain what `nolint` means.

File: Mathlib/Data/List/Basic.lean [hunk 3]
```diff
@@ -1100,25 +1100,6 @@ theorem indexOf_nil (a : α) : indexOf a [] = 0 :=
...
-  where
-    findIdx_go_succ (p : α → Bool) (l : List α) (n : ℕ) :
-        List.findIdx.go p l (n + 1) = (List.findIdx.go p l n) + 1 := by
-      cases l with
-      | nil => unfold List.findIdx.go; exact Nat.succ_eq_add_one n
-      | cons head tail =>
-        unfold List.findIdx.go
-        cases p head <;> simp only [cond_false, cond_true]
-        exact findIdx_go_succ p tail (n + 1)
-
```
Comment (eric-wieser) [inline Mathlib/Data/List/Basic.lean:1121]: I assume these were upstreamed; please mention that (maybe just "all the deletions have been upstreamed") in the PR description

File: Mathlib/Data/Nat/Digits.lean [hunk 4]
```diff
@@ -172,10 +172,10 @@ theorem ofDigits_eq_foldr {α : Type*} [Semiring α] (b : α) (L : List ℕ) :
 #align nat.of_digits_eq_foldr Nat.ofDigits_eq_foldr
 
 theorem ofDigits_eq_sum_map_with_index_aux (b : ℕ) (l : List ℕ) :
-    ((List.range l.length).zipWith ((fun i a : ℕ => a * b ^ i) ∘ succ) l).sum =
+    ((List.range l.length).zipWith ((fun i a : ℕ => a * b ^ (i + 1))) l).sum =
```
Comment (eric-wieser) [inline Mathlib/Data/Nat/Digits.lean:175]: Can you explain what happened here in the PR description?

**Review by eric-wieser — APPROVED** [review]
bors d+

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**